src/datafactory/azext_datafactory/generated/_params.py

```diff
@@ -0,0 +1,633 @@
+                   'uld match existing entity or can be * for unconditional update.')
+        c.argument('properties', arg_type=CLIArgumentType(options_list=['--properties'], help='Dataset properties. Expe'
+                                                          'cted value: json-string/@json-file.'))
+
+    with self.argument_context('datafactory dataset delete') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('factory_name', help='The factory name.', id_part='name')
+        c.argument('dataset_name', options_list=['--name', '-n'], help='The dataset name.', id_part='child_name_1')
+
+    with self.argument_context('datafactory pipeline list') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('factory_name', help='The factory name.')
+
+    with self.argument_context('datafactory pipeline show') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('factory_name', help='The factory name.', id_part='name')
+        c.argument('pipeline_name', options_list=['--name', '-n'], help='The pipeline name.', id_part='child_name_1')
+        c.argument('if_none_match', help='ETag of the pipeline entity. Should only be specified for get. If the ETag ma'
+                   'tches the existing entity tag, or if * was provided, then no content will be returned.')
+
+    with self.argument_context('datafactory pipeline create') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('factory_name', help='The factory name.')
+        c.argument('pipeline_name', options_list=['--name', '-n'], help='The pipeline name.')
+        c.argument('if_match', help='ETag of the pipeline entity.  Should only be specified for update, for which it sh'
+                   'ould match existing entity or can be * for unconditional update.')
+        c.argument('pipeline', arg_type=CLIArgumentType(options_list=['--pipeline'], help='Pipeline resource definition'
+                                                        '. Expected value: json-string/@json-file.'))
+
+    with self.argument_context('datafactory pipeline update') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('factory_name', help='The factory name.', id_part='name')
+        c.argument('pipeline_name', options_list=['--name', '-n'], help='The pipeline name.', id_part='child_name_1')
+        c.argument('if_match', help='ETag of the pipeline entity.  Should only be specified for update, for which it sh'
+                   'ould match existing entity or can be * for unconditional update.')
+        c.argument('pipeline', arg_type=CLIArgumentType(options_list=['--pipeline'], help='Pipeline resource definition'
+                                                        '. Expected value: json-string/@json-file.'))

```

Comment:
In `CLIArgumentType`, if you don't rename the option, the  you don't need to set `options_list`.

Reply:
This should have been resolved now. Please review it again. Thx